airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -302,6 +306,10 @@ def update_dag_run_state(dag_id: str, dag_run_id: str, session) -> dict:
         raise BadRequest(detail=str(err))
 
     state = post_body['state']
-    dag_run.set_state(state=DagRunState(state))
-    session.merge(dag_run)
+    dag = current_app.dag_bag.get_dag(dag_id)
+    if state == State.SUCCESS:
+        set_dag_run_state_to_success(dag, dag_run.execution_date, commit=True)
+    else:
+        set_dag_run_state_to_failed(dag, dag_run.execution_date, commit=True)
```

Comment:
```suggestion
    elif state == State.FAILED:
    	set_dag_run_state_to_failed(dag, dag_run.execution_date, commit=True)
    else:
    	raise BadRequest(f"Bad value for `state`: {state}")
```

If the `state` field in the post body is anything other than `"success"`, this will cause the DAGRun to be marked as failed.

For example, if someone tries to update a DAGRun with `{state: succes}` it will mark the DAGRun as failed due to the spelling error.

I think its a lot safer to do nothing in the event of an unexpected or input.

Reply:
We have a check in the open api documentation that won’t allow any other state apart from `success` and `failed` State to get to this endpoint

Reply:
You can check the test cases for this endpoint

Reply:
My mistake, I didn't know about - Thanks for the explanation.